src/Orleans.Serialization/Codecs/IntegerCodec.cs

```diff
@@ -745,4 +749,210 @@ public static ulong ReadValue<TInput>(ref Reader<TInput> reader, Field field)
+        /// <param name="expectedType">The expected type.</param>
+        /// <param name="value">The value.</param>
+        public static void WriteField<TBufferWriter>(ref Writer<TBufferWriter> writer, uint fieldIdDelta, Type expectedType, UInt128 value) where TBufferWriter : IBufferWriter<byte>
+        {
+            if (value <= (UInt128)ulong.MaxValue && value >= (UInt128)ulong.MinValue)
+            {
+                UInt64Codec.WriteField(ref writer, fieldIdDelta, expectedType, (ulong)value, CodecFieldType);
+            }
+            else
+            {
+                ReferenceCodec.MarkValueField(writer.Session);
+                UnsignedBinaryIntegerCodec<UInt128>.WriteField(ref writer, fieldIdDelta, expectedType, in value);
+            }
+        }
+
+        /// <inheritdoc/>
+        UInt128 IFieldCodec<UInt128>.ReadValue<TInput>(ref Reader<TInput> reader, Field field) => ReadValue(ref reader, field);
+
+        /// <summary>
+        /// Reads a value.
+        /// </summary>
+        /// <typeparam name="TInput">The reader input type.</typeparam>
+        /// <param name="reader">The reader.</param>
+        /// <param name="field">The field.</param>
+        /// <returns>The value.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static UInt128 ReadValue<TInput>(ref Reader<TInput> reader, Field field)
+        {
+            if (field.WireType != WireType.LengthPrefixed)
+            {
+                return UInt64Codec.ReadValue(ref reader, field);
+            }
+
+            ReferenceCodec.MarkValueField(reader.Session);
+            UInt128 result;
+            UnsignedBinaryIntegerCodec<UInt128>.ReadLengthPrefixed(ref reader, out result);
+            return result;
+        }
+
+        /// <summary>
+        /// Reads a value without protocol framing.
+        /// </summary>
+        /// <typeparam name="TInput">The reader input type.</typeparam>
+        /// <param name="reader">The reader.</param>
+        /// <returns>The value.</returns>
+        internal static UInt128 ReadRaw<TInput>(ref Reader<TInput> reader)
+        {
+            UInt128 result;
+            UnsignedBinaryIntegerCodec<UInt128>.ReadLengthPrefixed(ref reader, out result);
+            return result;
+        }
+    }
+
+    internal static class SignedBinaryIntegerCodec<T> where T : struct, IBinaryInteger<T>, ISignedNumber<T>
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void WriteField<TBufferWriter>(scoped ref Writer<TBufferWriter> writer, uint fieldIdDelta, Type expectedType, scoped in T value) where TBufferWriter : IBufferWriter<byte>
+        {
+            var byteCount = value.GetByteCount();
+            writer.WriteFieldHeader(fieldIdDelta, expectedType, typeof(T), WireType.LengthPrefixed);
+            writer.WriteVarUInt32((uint)byteCount);
+            Span<byte> bytes = byteCount < 64 ? stackalloc byte[byteCount] : new byte[byteCount];
```

Comment:
Perf-wise stack-allocating to a constant (power of 2) value is best, especially together with `SkipLocalsInit`, as it boils down to just a stack-pointer move + the stack-cookie (safety guard).

See [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBkIYBvGgzNNdxFAwCy5ABS8eDYAE8MMEdwwBKBqfMTanMQhlkxbC4AHjcPAD4Xd1UGAF4GXAxsMABrbElJCDBEjwBtWM8IbwBdDWDzAF8AsyaLJmsbUkdncq8ePxag0PNwyJikhPLcVPTMnLyCovKS1CqWWUllGHs6NGKK7x9akMa65tOGEtls3jEAGUK83ABJLl4MKpadNtttLow93q+fznQZDMIRaLlCZJKZpDJZXL5Qp7MpJQE1FonEItS7XO4PSTPV7vT6WdooP4Ayp9YEhUFDEaQ8Z7WEzBHzZFLFZrDYqba7HrU3xHBo0epAA=) (`M4` is best).

So
```suggestion
            Span<byte> bytes = byteCount < 64 ? stackalloc byte[64].Slice(byteCount) : new byte[byteCount];
```
and add the `[SkipLocalsInit]` attribute either to the function, class or assembly (if it's safe to assume to no usage relies on cleared (zeroed) locals).

Reply:
Or leave it as is here and do a follow-up pass to audit all of the stack-allocs?
(I can have a look a bit later into this, if needed)